app/lib/features/pins/widgets/pin_item.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/common/widgets/html_editor.dart';
+import 'package:acter/features/home/widgets/space_chip.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' show ActerPin;
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class PinItem extends ConsumerStatefulWidget {
+  final ActerPin pin;
+  final ScrollController? controller;
+  const PinItem(this.pin, this.controller, {super.key});
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _PinItemState();
+}
+
+class _PinItemState extends ConsumerState<PinItem> {
+  late TextEditingController _linkController;
+  final GlobalKey<FormState> _formkey = GlobalKey<FormState>();
+
+  @override
+  void initState() {
+    super.initState();
+    _linkController = TextEditingController(text: widget.pin.url() ?? '');
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final pinEdit = ref.watch(pinEditStateProvider(widget.pin));
+    final pinEditNotifier =
+        ref.watch(pinEditStateProvider(widget.pin).notifier);
+    final isLink = widget.pin.isLink();
+    final spaceId = widget.pin.roomIdStr();
+
+    final List<Widget> content = [
+      Container(
+        alignment: Alignment.topLeft,
+        margin: const EdgeInsets.symmetric(vertical: 8),
+        child: SpaceChip(spaceId: spaceId),
+      ),
+    ];
+
+    if (isLink) {
+      content.add(
+        Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: TextFormField(
+            controller: _linkController,
+            readOnly: !pinEdit.editMode,
+            style: Theme.of(context).textTheme.bodyMedium,
+            decoration: InputDecoration(
+              prefixIcon: IconButton(
+                onPressed: () async => await openLink(pinEdit.link, context),
+                icon: const Icon(Atlas.link_chain_thin),
+                iconSize: 18,
+              ),
+            ),
+            validator: (value) {
+              if (value != null) {
+                final uri = Uri.tryParse(value);
+                if (uri == null || !uri.isAbsolute) {
+                  return 'link is not valid';
+                }
+              }
+              return null;
+            },
+          ),
```

Comment:
I am not quite sure how this works now. the user has to click the tiny icon on the left to open the link? I am pretty sure in more cases than not they want to open the link when they click it. not switch into an edit-mode ... if anything the UX should be the other way around: show a suffix-edit icon if the user can edit and only upon clicking that switch into edit-mode and on all other clicks, open the link...

Reply:
The user can click on icon to launch url link, yes. Clicking on field will not switch it in edit mode as user has to specifically go in pin item actions to enable editing (which makes every field editable, then).